torch/jit/_script.py

```diff
@@ -342,15 +343,19 @@ def unpackage_script_module(importer: PackageImporter, script_module_id: str) ->
     Called by ``torch.package.PackageImporter``'s Pickler's ``persistent_load`` function.
     Performs work of loading and returning a ScriptModule from a ``torch.package`` archive.
     """
-    if not isinstance(importer.zip_reader, torch._C.PyTorchFileReader):
+
+    # if 'storage_context' not in importer.external_registry:
+    #     importer.external_registry['storage_context'] = torch._C.DeserializationStorageContext()
+
+    if not isinstance(importer.zip_reader, TorchScriptPackageZipFileReader):
         raise RuntimeError(
             "Loading ScriptObjects from a PackageImporter created from a "
-            "directory is not supported. Use a package archive file instead."
+            f"directory is not supported. Use a package archive file instead. is of type {type(importer.zip_reader)}"
         )
     cu = torch._C.CompilationUnit()
     cpp_module = torch._C._import_ir_module_from_package(
         cu,
-        importer.zip_reader,
+        importer.zip_reader.zip_file_reader,
```

Comment:
does this _need_ the C++ implementation? wondering if using the abstract classes here is just adding more layers without providing much simplification

Reply:
Yeah unfortunately, I think it does. At least I haven't figured out a nice way to avoid invoking it.